nifi-extension-bundles/nifi-box-bundle/nifi-box-processors/src/main/java/org/apache/nifi/processors/box/UpdateBoxMetadataTemplate.java

```diff
@@ -0,0 +1,464 @@
+        if (hiddenObj != null) {
+            field.hidden = Boolean.parseBoolean(hiddenObj.toString());
+        }
+
+        final Object descriptionObj = record.getValue("description");
+        if (descriptionObj != null) {
+            field.description = descriptionObj.toString();
+        }
+
+        // For enum or multiSelect fields, capture options
+        if (("enum".equals(type) || "multiSelect".equals(type))) {
+            final Object optionsObj = record.getValue("options");
+            if (optionsObj instanceof List<?> optionsList) {
+                field.options = optionsList.stream()
+                        .filter(Objects::nonNull)
+                        .map(Object::toString)
+                        .collect(Collectors.toList());
+            }
+        }
+
+        fields.add(field);
+        processedKeys.add(key);
+    }
+
+    private List<MetadataTemplate.FieldOperation> generateOperations(final MetadataTemplate existingTemplate,
+                                                                     final List<FieldDefinition> desiredFields) {
+        final Map<String, MetadataTemplate.Field> existingFieldsByKey =
+                existingTemplate.getFields().stream()
+                        .collect(Collectors.toMap(MetadataTemplate.Field::getKey, f -> f));
+
+        final List<MetadataTemplate.FieldOperation> operations = new ArrayList<>();
+
+        for (FieldDefinition desiredField : desiredFields) {
+            final MetadataTemplate.Field existingField = existingFieldsByKey.get(desiredField.key);
+
+            if (existingField == null) {
+                // Field doesn't exist - add it
+                operations.add(createAddFieldOperation(desiredField));
+            } else {
+                // Field exists - check if it needs updating
+                final Map<String, Object> changes = getFieldChanges(existingField, desiredField);
+                if (!changes.isEmpty()) {
+                    operations.add(createEditFieldOperation(existingField.getKey(), changes));
+                }
+                existingFieldsByKey.remove(desiredField.key);
+            }
+        }
+
+        // Any leftover fields in existingFieldsByKey are not desired - remove them
+        for (final String keyToRemove : existingFieldsByKey.keySet()) {
+            operations.add(createRemoveFieldOperation(keyToRemove));
+        }
+        return operations;
+    }
+
+    private Map<String, Object> getFieldChanges(final MetadataTemplate.Field existingField,
+                                                final FieldDefinition desiredField) {
+        final Map<String, Object> changes = new HashMap<>();
+
+        if (!existingField.getKey().equals(desiredField.key)) {
+            changes.put("key", desiredField.key);
+        }
+
+        if (desiredField.displayName != null && (existingField.getDisplayName() == null
+                || !existingField.getDisplayName().equals(desiredField.displayName))) {
```

Comment:
Nit: it can be simplified.
```suggestion
        if (desiredField.displayName != null && !desiredField.displayName.equals(existingField.getDisplayName())) {
```